exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKitExtension.java

```diff
@@ -29,22 +29,88 @@
 import org.junit.jupiter.api.extension.ParameterResolutionException;
 import org.junit.jupiter.api.extension.ParameterResolver;
 
+/**
+ * Extension that injects TestKit into service tests. Register this extension with TestKit builder
+ * and a TestKit will be injected as a parameter, instantiated from given builder.
+ *
+ * Allows injecting TestKit into test context:
+ *
+ * <pre><code>
+ * @RegisterExtension
+ * static TestKitExtension testKitExtension = new TestKitExtension(
+ *     TestKit.builder()
+ *         .withService(TestServiceModule.class));
+ *
+ * @BeforeEach
+ * void setUp(TestKit testKit) {
+ *   // Set up logic
+ * }
+ *
+ * @Test
+ * void test(TestKit testKit) {
+ *   // Test logic
+ * }
+ * </code></pre>
+ *
+ * instead of:
+ *
+ * <pre><code>
+ * static {
+ *   LibraryLoader.load();
+ * }
+ *
+ * @Test
+ * void test() {
+ *   try (TestKit testKit = TestKit.forService(TestServiceModule.class)) {
+ *     // Set up logic
+ *     // Test logic
+ *   }
+ * }
+ * </code></pre>
+ *
+ * <p>As different tests might need slightly different TestKit configuration, following
+ * parameterization annotations are available:
+ * <ul>
+ *   <li>{@link Validator} sets main TestKit validator node type to validator</li>
```

Comment:
Do we need validator? Won't you usually use validator by default, and configure Auditors if needed? Or keep it for completeness?

Reply:
I think we should have both. It's possible that a user needs most of his tests to work with an auditor TestKit, but a couple to have a validator main node.

We could also go with a single annotation `@NodeType(EmulatedNodeType.VALIDATOR/AUDITOR)`.